sdk/python/kfp/dsl/component_factory.py

```diff
@@ -132,6 +133,14 @@ def make_pip_install_command(
 PIP_DISABLE_PIP_VERSION_CHECK=1 {pip_install_commands} && "$0" "$@"
 '''
 
+# Creates and activates a virtual environment in a temporary directory.
+# The environment inherits the system site packages.
+_use_venv_script_template = '''
+tmp=$(mktemp -d)
+python3 -m venv "$tmp/venv" --system-site-packages
+source "$tmp/venv/bin/activate"
```

Comment:
We might also need a cleanup mechanism to ensure temporary directories used for virtual environments are deleted after use.

Reply:
Thanks!

A few questions:

* Is it a must-have or a nice-to-have? 
* What's the rationale for the cleanup? The main executor does not have such mechanism AFAIK. 
* Isn't the life-time of the pod sufficient to handle cleanup?